Vespa 8 changes (#502)
vespa-engine/cloud: en/vespa8-release-notes.md

layout: page
---

{% include note.html content="Vespa 8 will be released in June 2022."%}

When Vespa 8 is released, it will become available on Vespa Cloud.
Applications control when they start using the new major version
(see below), and must do so within three months after the release.
This document lists changes between Vespa 7 and 8 that are specific for Vespa Cloud
as well as the steps needed to start using it.
You should also read the [Vespa 8 release notes](https://docs.vespa.ai/en/vespa8-release-notes.html).

## Changes in deployment.xml

### Changed syntax for global endpoints

The `global-service-id` attribute in the `<prod>` element has been removed, along with
the `active` attribute in the `<region>` element. Instead, global endpoints are
now configured with the new `<endpoints>` element. See the 
[Migration guide](https://cloud.vespa.ai/en/reference/routing.html#deprecated-syntax) 
for details.

## Procedure to move an application to Vespa 8

- Ensure there are no deprecation warnings in the application notifications in the console.

> - If you application has Java components: Ensure there are no deprecation warnings when compiling it.

Reviewer: ```suggestion
- If your application has Java components: Ensure there are no deprecation warnings when compiling it.
```

- Review the other changes listed in the [Vespa 8 release notes](https://docs.vespa.ai/en/vespa8-release-notes.html) and make
any necessary changes to your application.
- Set <code>major-version=8</code> on the <a href="reference/deployment.html#deployment">&lt;deployment&gt;</a> 
tag in your deployment.xml file.

> To try out running your application on Vespa 8 in <code>dev</code> you can pass <code>-Dvespa.compile.version</code>,

Reviewer: ```suggestion
Deployments to <code>dev</code> will default to Vespa 8 once released.
To stay on Vespa 7, specify a version when deploying through the Vespa Cloud UI,
or use <code>vespa deploy --version &lt;some 7 version&gt;</code> with Vespa CLI.
If your application has Java components, instead use <code>mvn package -Dvespa.compile.version=&lt;some 7 version&gt;</code>
when building it. (This will ensure a compatible runtime.)
```

Reviewer: Like so?

now configured with the new `<endpoints>` element. See the 
[Migration guide](https://cloud.vespa.ai/en/reference/routing.html#deprecated-syntax) 
for details.

## Procedure to move an application to Vespa 8

- Ensure there are no deprecation warnings in the application notifications in the console.
- If your application has Java components: Ensure there are no deprecation warnings when compiling it.
- Review the other changes listed in the [Vespa 8 release notes](https://docs.vespa.ai/en/vespa8-release-notes.html) and make
any necessary changes to your application.
- Set <code>major-version=8</code> on the <a href="reference/deployment.html#deployment">&lt;deployment&gt;</a> 
tag in your deployment.xml file.

Deployments to <code>dev</code> will default to Vespa 8 once released.
To stay on Vespa 7, specify a version when deploying through the Vespa Cloud UI,
or use <code>vespa deploy --version &lt;some 7 version&gt;</code> with Vespa CLI.
If your application has Java components, instead use <code>mvn package -Dvespa.compile.version=&lt;some 7 version&gt;</code>
when building it. (This will ensure a compatible runtime.)

> with some valid Vespa 8 version.

Reviewer: ```suggestion
```

Remove sstray line
